packages/rich-text/src/component/index.js

```diff
@@ -260,7 +260,7 @@ class RichText extends Component {
 		if ( onPaste ) {
 			// Only process file if no HTML is present.
 			// Note: a pasted file may have the URL as plain text.
-			const image = find( [ ...items, ...files ], ( { type } ) =>
+			const images = filter( [ ...items, ...files ], ( { type } ) =>
```

Comment:
This array merging is probably the cause of the duplicated images. We should make sure there are no duplicates, or use one or the other (if one doesn't exist).

Reply:
I think the safest is to still use both, but discard any duplicates.

Reply:
Looks like you'll have to compare the `name` and `size`.

Reply:
@ellatrix I was thinking on trimming the array to size 1, but if you can clear the duplicates it will be even better! Feel free to push to the PR :)

Reply:
Done! :)